source/core/owt_base/AudioFrameConstructor.cpp

```diff
@@ -47,6 +51,52 @@ int AudioFrameConstructor::deliverVideoData_(std::shared_ptr<erizo::DataPacket>
     return 0;
 }
 
+class AudioLevel {
+ public:
+  uint32_t ext_info:8;
+  uint8_t al_data:8;
+
+  inline uint8_t getId() {
+    return ext_info >> 4;
+  }
+  inline uint8_t getLength() {
+    return (ext_info & 0x0F);
+  }
+  inline bool getVoice() {
+    return (al_data & 0x80) != 0;
+  }
+  inline uint8_t getLevel() {
+    return al_data & 0x7F;
+  }
+};
+
+AudioLevel* parseAudioLevel(std::shared_ptr<erizo::DataPacket> p) {
+    const erizo::RtpHeader* head = reinterpret_cast<const erizo::RtpHeader*>(p->data);
+    AudioLevel* ret = nullptr;
+    if (head->getExtension()) {
+        uint16_t totalExtLength = head->getExtLength();
+        if (head->getExtId() == 0xBEDE) {
+            char* extBuffer = (char*)&head->extensions;  // NOLINT
+            uint8_t extByte = 0;
+            uint16_t currentPlace = 1;
+            uint8_t extId = 0;
+            uint8_t extLength = 0;
+            while (currentPlace < (totalExtLength*4)) {
+                extByte = (uint8_t)(*extBuffer);
+                extId = extByte >> 4;
+                extLength = extByte & 0x0F;
+                if (extId == kAudioLevelExtensionId) {
+                    ret = reinterpret_cast<AudioLevel*>(extBuffer);
```

Comment:
Not good to return a struct ptr created from another struct's member. Please copy value over as you're not sure if anybody is going to change the input p's content.